pkg/server/api.go

```diff
@@ -115,11 +123,15 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	data, err := json.Marshal(conf)
+	data, translated, err := translateConfigForIgnitionMiddleware(conf, useragent)
```

Comment:
To make sure I am understanding this bit correctly, this is what happens when a node's ignition makes a request towards the MCS port: we hit this function, which passes effectively the rendered-config for the pool in ignition format to the node. This translation will essentially compare the ignition version being requested and the ignition version of the rendered config, and translate to match them. Today the rendered config is v2 so we won't see `ConvertIgnition2to3` being used

Reply:
yes, that's exactly right! (we can't update the rendered ign to v3 right away, as the MCD changes need to land in the RPM in RHCOS first). Should I add commentary here?

Reply:
> as the MCD changes need to land in the RPM in RHCOS first

This might not be true anymore right? #1766 cc @cgwalters

Reply:
Sorry I'm not sure if I completely follow. The request should have come from ignition itself, and we're passing the data for the ignition binary to parse. So the MCD RPM doesn't touch this flow, right?

Reply:
Right, in thta case MCD is not involved: Serving spec 3 to nodes requesting it will work with this PR.

However the internal state/rendered MC will still be saved as spec v2 here, as our MCD doesn't understand spec 3 yet (for Day 2 ops). If we can merge in #1766 first, it will make things much clearer and saner.

Reply:
> Should I add commentary here?

YES PLEASE!

Reply:
wait that pr merged...

Reply:
added a comment here for clarification